### PR TITLE
feat: Unify dialogs across app - part 1

### PR DESCRIPTION
**Pull request by lrupnikk**

### CODE REVIEW

**Review by dabitdev — APPROVED** [review]

**Comment by dabitdev** [comment]
Looks like is not compiling under test tasks.

```
w: Some runtime JAR files in the classpath have an incompatible version. Consider removing them from the classpath
e: /home/circleci/brd-mobile/brd-android/app/src/main/java/com/breadwallet/ui/home/HomeController.kt: (84, 42): Unresolved reference: AlertDialogController

> Task :brd-android:app:compileDevelopmentTestnetDebugKotlin FAILED
FAILURE: Build failed with an exception
```

File: brd-android/app/src/main/java/com/breadwallet/ui/navigation/NavigationTarget.kt [hunk 1]
```diff
@@ -80,7 +80,8 @@ sealed class NavigationTarget : INavigationTarget {
 
     data class FabriikToast(
         val type: Type,
-        val message: String
+        val message: String? = null,
+        val messageRes: Int? = null
```
Comment (dabitdev) [inline brd-android/app/src/main/java/com/breadwallet/ui/navigation/NavigationTarget.kt]: could you add here StringRes annotation? to make sure the int comes from String resources?

File: brd-android/fabriik-common/src/main/java/com/fabriik/common/ui/dialog/FabriikGenericDialog.kt [hunk 2]
```diff
@@ -81,29 +82,39 @@ class FabriikGenericDialog : DialogFragment() {
...
         button.isVisible = args.showDismissButton
         button.setOnClickListener {
-            notifyListeners(RESULT_KEY_DISMISSED)
+            notifyListeners(RESULT_KEY_DISMISSED, null)
         }
     }
 
-    private fun notifyListeners(result: String) {
+    private fun setupInputTextField(view: TextInputLayout) {
+        val hintText = if (args.textInputHintRes != null) getString(args.textInputHintRes!!) else args.textInputHint
```
Comment (roknovak) [inline brd-android/fabriik-common/src/main/java/com/fabriik/common/ui/dialog/FabriikGenericDialog.kt]: Do we really need Input field here?

File: brd-android/app/src/main/java/com/breadwallet/ui/home/HomeScreen.kt [hunk 3]
```diff
@@ -182,12 +187,21 @@ object HomeScreen {
...
-                titleResId = R.string.SupportForm_helpUsImprove,
-                messageResId = R.string.SupportForm_feedbackAppreciated,
-                positiveButtonResId = R.string.Button_submit,
-                negativeButtonResId = R.string.SupportForm_notNow,
-                textInputPlaceholderResId = R.string.SupportForm_pleaseDescribe
+            override val navigationTarget = NavigationTarget.FabriikGenericDialog(
+                FabriikGenericDialogArgs(
+                    titleRes = R.string.SupportForm_helpUsImprove,
+                    descriptionRes = R.string.SupportForm_feedbackAppreciated,
+                    textInputHintRes = R.string.SupportForm_pleaseDescribe,
```
Comment (roknovak) [inline brd-android/app/src/main/java/com/breadwallet/ui/home/HomeScreen.kt]: I think we do not need an input view inside the dialog. This effect could be changed and we can show feedback in similar way as we do on the Menu screen. Here we can just replace positive button text with Continue and open Email client when user click on it

**Review by roknovak — APPROVED** [review]

**Review by dabitdev — APPROVED** [review]
LGTM

**Pull request merged**